src/api/src/main.py

```
@@ -231,17 +335,17 @@ async def iterfile(file_size):
                 await asyncio.sleep(sleep_duration)  # throttle download speed
                 yield chunk
 
-        file_size = book_buffer.getbuffer().nbytes
+        file_size = output_buffer.getbuffer().nbytes
 
         return StreamingResponse(
             iterfile(file_size),
             media_type=media_type,
             headers={
-                "Content-Disposition": f'attachment; filename="{slugify(metadata["title"])}_{story_id}{"_images" if download_images else ""}.{format.value}"',  # Thanks https://stackoverflow.com/a/72729058
+                "Content-Disposition": f'attachment; filename="{slugify(metadata["name" if mode==DownloadMode.list else "title"]) if id_download else (username+'_'+("archive" if mode is DownloadMode.archive else "library"))}{'_'+str(download_id) if id_download else ""}{"_images" if download_images else ""}{'_'+format.value if extension is "zip" else ""}.{extension}"',  # Thanks https://stackoverflow.com/a/72729058
```

Ruff (0.15.11)

[error] Use `==` to compare constant literals

Replace `is` with `==`

(F632)